fundingmanager.go

```diff
@@ -1600,47 +1702,124 @@ func (f *fundingManager) sendFundingLocked(completeChan *channeldb.OpenChannel,
 	return nil
 }
 
-// sendChannelAnnouncement broadcast the necessary channel announcement
-// messages to the network. Should be called after the fundingLocked message
-// is sent (channelState is 'fundingLockedSent') and the channel is ready to
-// be used.
-func (f *fundingManager) sendChannelAnnouncement(completeChan *channeldb.OpenChannel,
-	channel *lnwallet.LightningChannel, shortChanID *lnwire.ShortChannelID) error {
-
-	// TODO(eugene) wait for 6 confirmations here
+// addToRouterGraph sends a ChannelAnnouncement and a ChannelUpdate to the
+// gossiper so that the channel is added to the Router's internal graph.
+// These announcement messages are NOT broadcasted to the greater network,
+// only to the channel counter party. The proofs required to announce the
+// channel to the greater network will be  created and sent in annAfterSixConfs.
+func (f *fundingManager) addToRouterGraph(completeChan *channeldb.OpenChannel,
+	channel *lnwallet.LightningChannel,
+	shortChanID *lnwire.ShortChannelID) error {
 
 	chanID := lnwire.NewChanIDFromOutPoint(&completeChan.FundingOutpoint)
-	fundingPoint := completeChan.FundingOutpoint
 
-	fndgLog.Infof("Announcing ChannelPoint(%v), short_chan_id=%v",
-		&fundingPoint, spew.Sdump(shortChanID))
-
-	// Register the new link with the L3 routing manager so this new
-	// channel can be utilized during path finding.
-	err := f.announceChannel(f.cfg.IDKey, completeChan.IdentityPub,
-		channel.LocalFundingKey, channel.RemoteFundingKey,
-		*shortChanID, chanID)
+	ann, err := f.newChanAnnouncement(f.cfg.IDKey, completeChan.IdentityPub,
+		channel.LocalFundingKey, channel.RemoteFundingKey, *shortChanID,
+		chanID)
 	if err != nil {
-		return fmt.Errorf("channel announcement failed: %v", err)
+		return fmt.Errorf("error generating channel "+
+			"announcement: %v", err)
+	}
+
+	// Send ChannelAnnouncement and ChannelUpdate to the gossiper to add
+	// to the Router's topology.
+	if err = f.cfg.SendAnnouncement(ann.chanAnn); err != nil {
+		return fmt.Errorf("error channel announcement: %v", err)
+	}
+	if err = f.cfg.SendAnnouncement(ann.chanUpdateAnn); err != nil {
```

Comment:
This will need to check the returned error. As atm, it may be rejected if this function _partially_ succeeded. It should be looking for `routing.ErrOutdated` or `routing.ErrIgnored`.

Reply:
The other instance (where we actually try to announce), should also be modified to check the error. Otherwise, if we partially process the state, then we may never actually update the state in the database.